### ISSUE TITLE
Test refactors should utilize a TDD approach 

### ISSUE THREAD

**Issue by jonbrenner** [quii/learn-go-with-tests#589]
When we refactor our tests in the book, we often begin by implementing new structs/functions before using them in the test cases. For example, see the [Still Some Problems](https://github.com/quii/learn-go-with-tests/blob/main/mocking.md#still-some-problems) section in the Mocking chapter. Here we implement the new spies for checking the order of operations prior to using them in the `sleep before every print` test.

In my opinion, we should continue to use a TDD approach in refactors by specifying the usage of the new code before the implementation of it. In other words, we use write the `sleep before every print` test first and then let the errors guide us to the implementation of the `SpyCountdownOperations` spy. This "specify, then implement" process reinforces the notion that TDD is a design technique as much as a testing technique.

**Comment by quii** [comment]
Fair point, and I get the appeal of staying strict about "specify, then implement" throughout. My take though: spies/test-doubles are scaffolding around the code under test, not the production code TDD is meant to be driving out — so introducing them first is a deliberate choice to keep the narrative focused on the actual behaviour being tested, rather than the mechanics of building the double. Closing this one, but appreciate the thoughtful writeup!
